pkg/dmi.go

```diff
@@ -0,0 +1,169 @@
+package pkg
+
+import (
+	"bufio"
+	"fmt"
+	"os/exec"
+	"strings"
+)
+
+type Entry = map[string][]string
+type DMIMap = map[string]Entry
+
+type DMIDecoder struct {
+	installed  bool
+	cmdOutput  string
+	decodedMap DMIMap
+}
+
+// NewCheckerService creates new instance from the parser
+func NewDMIDecoder() DMIDecoder {
+	dmidecode := exec.Command("sudo", "dmidecode")
+	res, _ := dmidecode.Output()
+
+	if len(res) > 0 {
+		return DMIDecoder{
+			installed:  true,
+			cmdOutput:  string(res),
+			decodedMap: DMIMap{},
+		}
+	} else {
+		fmt.Println("dmidecode is not installed in your device")
```

Comment:
this should be an error to be returned, if for logging let's use `zerolog` or standard go logging